web_ui/src/pages/project-details/components/project-labels/project-labels.component.tsx

```diff
@@ -29,18 +35,23 @@ export const ProjectLabels = (): JSX.Element => {
     const { editProjectLabelsMutation } = useProjectActions();
     const datasetIdentifier = useDatasetIdentifier();
 
-    const { project, isTaskChainProject, reload } = useProject();
+    const { project, isTaskChainProject, reload, isSingleDomainProject } = useProject();
 
     const [isDirty, setIsDirty] = useState<boolean>(false);
     const [isDialogOpen, setDialogOpen] = useState<boolean>(false);
     const [isEditionEnabled, setEditionEnablement] = useState<boolean>(false);
     const [tasksMetadata, setTasksMetadata] = useState<TaskMetadata[]>(getTasksMetadata(project.tasks));
     const [labelsValid, setLabelsValidity] = useState<boolean>(true);
+    const [saveButtonDisabled, setSaveButtonDisabled] = useState<boolean>(false);
 
     const { addNotification } = useNotification();
 
     const [isOpen, setIsOpen, onUnsavedAction] = useHistoryBlock(isEditionEnabled);
 
+    useEffect(() => {
+        setSaveButtonDisabled(!labelsValid);
+    }, [labelsValid]);
+
```

Comment:
Can we avoid this by making `saveButtonDisbled` a computed variable? E.g.,
```
const saveButtonDisabled = !labelsValid | hasAtLeastOneLabel(tasksMetadata)
```

where we could compute if there will be at least 1 label based on the task metadata instead?

Reply:
you know how much i like `useEffect` :sarcasm: , updated :)